src/vsr.zig

```diff
@@ -1525,6 +1529,30 @@ test "parse_addresses" {
     }
 }
 
+test "parse_addresses: fuzz" {
+    const test_count = 1024;
+    const len_max = 32;
+    const alphabet = " \t\n,:[]0123456789abcdefgABCDEFGXx";
+
+    const seed = std.crypto.random.int(u64);
+
+    var prng = std.rand.DefaultPrng.init(seed);
+    const random = prng.random();
+
+    var input_max: [len_max]u8 = .{0} ** len_max;
+    for (0..test_count) |_| {
+        const len = random.uintAtMost(usize, len_max);
+        var input = input_max[0..len];
+        for (input) |*c| {
+            c.* = alphabet[random.uintAtMost(usize, alphabet.len)];
+        }
+        if (parse_addresses(std.testing.allocator, input, 3)) |addresses| {
+            std.testing.allocator.free(addresses);
+        } else |_| {
```

Comment:
Does zig require this branch since it is an error?

Reply:
Yup, otherwise Zig assumes it's an option. I _think_ the idea here is to make error discards syntactically apparent --- `catch {}` and `else |_|` are the only forms that can discard errors (and they only discard errors). Something like `_ = err` can'd discard error. This is in contrast  to Rust, which doesn't distinguish between discarding errors and discarding any other value.

Reply:
but `zig fmt` disagrees with formatting here